client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/index.tsx

```diff
@@ -1,18 +1,26 @@
 import { LaunchpadContainer } from '@automattic/launchpad';
 import { StepContainer } from '@automattic/onboarding';
+import { getQueryArg } from '@wordpress/url';
 import React from 'react';
+import { useHostingProviderUrlDetails } from 'calypso/data/site-profiler/use-hosting-provider-url-details';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { HostingBadge } from './hosting-badge';
 import { Questions } from './questions';
 import { Sidebar } from './sidebar';
 import { SitePreview } from './site-preview';
 import type { Step } from '../../types';
 import './style.scss';
 
 const SiteMigrationInstructions: Step = function () {
-	const sidebar = <Sidebar />;
+	const importSiteQueryParam = getQueryArg( window.location.href, 'from' )?.toString() || '';
```

Comment:
+1

I think this change is important before merging because we have navigation without reload and it could cause issues. So it's important that the component reacts to the changes.